src/main/java/uk/gov/companieshouse/chs/gov/uk/notify/integration/api/service/GovUkNotifyService.java

```diff
@@ -50,11 +49,9 @@ public EmailResp sendEmail(
             SendEmailResponse response = client.sendEmail(templateId, recipient, personalisation, reference);
             return new EmailResp(response != null && response.getNotificationId() != null, response);
         } catch (NotificationClientException e) {
-            Map<String, Object> logData = Map.of(
-                    "recipient", recipient,
-                    "templateId", templateId
-            );
-            LOGGER.error("Failed to send email", e, new HashMap<>(logData));
+            Map<String, Object> logData = createLogData(reference);
```

Comment:
Statements at line 52 and 53 could have been combined to make it more meaningful e.g. 

Map.of(
            "reference", reference,
            "recipient", recipient,
            "templateId", templateId
    );

Could we directly pass the above statement in LOGGER.error method and get rid of "logData" variable ? It could make code more concise and readable.